### PR TITLE
Replace name speed_limit_composition with speed_limit_tags

### LINKED ISSUE #2435: The naming for speed limits per category is inconsistent
It couldn't be more all over the place:
 - tag in the back-end and django
 - category in django
 - composition in django / database table
 - at least one of the above in the front-end / editoast

We should:
 - [ ] pick one
 - [ ] make it consistent

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #2435

### CODE REVIEW

File: api/osrd_infra/migrations/0001_initial.py
Comment (flomonster) [inline api/osrd_infra/migrations/0001_initial.py]: You can't change an old migration. You must create a new one that actually changes the column name.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks! We should have a review from a front maintainer too.

**Review by eckter — DISMISSED** [review]
I haven't checked the details, but for variable renames, as long as the CI passes then it's good for me

**Review by flomonster — APPROVED** [review]

**Review by eckter — APPROVED** [review]

**Review by anisometropie — APPROVED** [review]

**Review by Math-R — APPROVED** [review]

**Pull request merged**
